.github/workflows/build.yml

```
@@ -19,4 +19,86 @@
+        $url = 'https://dev.azure.com/' + $organization + '/_apis/distributedtask/pools/' + $poolId + '/agents?includeCapabilities=true&api-version=7.1'
+
+        try {
+          # Get all agents
+          $response = Invoke-RestMethod -Uri $url -Method Get -Headers $headers
+        Write-Host "Successfully retrieved agents from pool $poolId"
+        } catch {
+            Write-Error "Failed to retrieve agents: $_"
+            exit 1
+        }
+
+        $agentCapabilities = $response.value | 
```

yamllint

[error] trailing spaces

(trailing-spaces)